lib/puppet/reports/http.rb

```diff
@@ -14,7 +14,7 @@
   def process
     url = URI.parse(Puppet[:reporturl])
     headers = { "Content-Type" => "application/x-yaml" }
-    options = {}
+    options = { :metric_id => [:puppet, :report, :http] }
```

Comment:
Got it. Thanks for clarifying @camlow325! In that case, yeah I think this key will basically get silently dropped in the current implementation, which seems just fine given your explanation.

Thought it might be risky to implicitly rely on `:metric_id` getting dropped, since `Puppet::Network::HTTP::Connection#post` is a bit removed from here. I'm not sure what the implications would be if we suddenly started including `:metric_id` in all report posts on accident, but we probably don't want to? 

It's not the cleanest thing but what do you think about checking if our `Puppet::Network::HttpPool::connection_class` is `Puppet::Server::HttpClient` instead of `Puppet::Network::Http::Connection` (or some other way of detecting that we're in the right environment) and only then adding `:metric_id` to the options? Do we do that kind of check anywhere else? Failing that, could you add a code comment with some explanation/warning that `:metric_id` is only used by the Puppet Server http client and is ignored otherwise?